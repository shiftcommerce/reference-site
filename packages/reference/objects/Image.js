import { Component } from 'react'
import classNames from 'classnames'

class Image extends Component {
  styles () {
    let basicStyles = {
      height: this.props.height,
      width: this.props.width
    }

    if (basicStyles.height < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.height * 19) / 20) }
    } else if (basicStyles.width < 200) {
      return { ...basicStyles, backgroundSize: ((basicStyles.width * 19) / 20) }
    } else {
      return { ...basicStyles, backgroundSize: 150 }
    }
  }

  render () {
    let {
      src,
      className
    } = this.props

    if (src) {
      return <img {...this.props} />
    } else {
      return <div className={classNames('c-dummy_image', className)} style={this.styles()} />
    }
  }
}

export default Image
