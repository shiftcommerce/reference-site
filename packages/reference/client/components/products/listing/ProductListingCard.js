// Libraries
import React from 'react'
import classNames from 'classnames'
import Link from 'next/link'

// Components
import ProductPrice from '../display/ProductPrice'

// Objects
import Image from '../../../objects/Image'

class ProductListingCard extends React.Component {
  renderRatingStars (product) {
    let rating = (product.rating || 0)

    return (
      [1, 2, 3, 4, 5].map((key, index) =>
        <span key={key} className={classNames({ 'c-product-listing-card__rating--fill': key <= rating, 'c-product-listing-card__rating--empty': key > rating })}>
          &#9733;
        </span>
      )
    )
  }

  render () {
    let {
      product
    } = this.props

    const assetFile = product.asset_files[0]

    return (
      <div className='c-product-listing-card'>
        <div className='c-product-listing-card__body'>
          <div className='c-product-listing-card__gallery  c-image'>
            <Link href={`/slug?slug=${product.canonical_path}`} as={product.canonical_path}>
              <Image className='c-product-listing-card__image'
                src={assetFile && assetFile.s3_url}
                alt={(assetFile && assetFile.alt_text) || product.title}
                aria-label={product.title} />
            </Link>
          </div>
          <div className='c-product-listing-card__title' >
            <p role='heading' aria-level='1'>
              {product.title}
            </p>
          </div>
          <div className='c-product-listing-card__price'>
            <ProductPrice variants={product.variants} />
          </div>
          <div className='c-product-listing-card__rating'>
            { this.renderRatingStars(product) }
          </div>
        </div>
      </div>
    )
  }
}

export default ProductListingCard
