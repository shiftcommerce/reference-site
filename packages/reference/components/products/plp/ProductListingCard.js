// Libraries
import React from 'react'
import classNames from 'classnames'
import Link from 'next/link'

// Components
import ProductPrice from '../pdp/ProductPrice'

// Objects
import Button from '../../../objects/Button'
import Image from '../../../objects/Image'

class ProductListingCard extends React.Component {
  renderRatingStars (product) {
    let rating = (product.rating || 0)

    return (
      [1, 2, 3, 4, 5].map((key, index) =>
        <span key={key} className={classNames({ 'c-product-listing-card__rating--fill': key <= rating, 'c-product-listing-card__rating--empty': key > rating })}>
          &#9733;
        </span>
      )
    )
  }

  render () {
    let {
      product
    } = this.props

    const assetFile = product.asset_files[0]

    return (
      <div className='c-product-listing-card'>
        <div className='c-product-listing-card__body'>
          <div className='c-product-listing-card__gallery  c-image'>
            <Image className='c-product-listing-card__image'
              src={assetFile && assetFile.s3_url}
              alt={(assetFile && assetFile.alt_text) || product.title}
              aria-label={product.title} />
          </div>
          <div className='c-product-listing-card__title' >
            <p role='heading' aria-level='1'>
              {product.title}
            </p>
          </div>

          <div className='c-product-listing-card__price'>
            <ProductPrice variants={product.variants} />
          </div>

          <div className='c-product-listing-card__view-details'>
            <Link href={`/slug?slug=${product.canonical_path}`} as={product.canonical_path}>
              <Button label='View' status='primary' size='lrg' aria-label='View' />
            </Link>
          </div>

          <div className='c-product-listing-card__rating'>
            {this.renderRatingStars(product)}
          </div>
        </div>
      </div>
    )
  }
}

export default ProductListingCard
