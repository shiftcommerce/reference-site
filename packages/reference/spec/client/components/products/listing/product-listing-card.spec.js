// Components
import ProductListingCard from '../../../../../client/components/products/listing/product-listing-card'

// Libraries
import Link from 'next/link'

// Objects
import Image from '../../../../../client/objects/image'

// Fixtures
import productSearchHit from '../../../../fixtures/product-search-hit'

test('renders ProductListingCard correctly', () => {
  // Act
  const wrapper = mount(
    <ProductListingCard
      title={productSearchHit.product_title}
      assetFileUrl={productSearchHit.product_assets[0].url}
      assetFileAltText={productSearchHit.product_assets[0].alt_text}
      minPrice={productSearchHit.variant_meta_data.eu.price}
      maxPrice={productSearchHit.variant_meta_data.eu.price}
      productPath={productSearchHit.product_path}
      productRating={productSearchHit.product_rating}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toContainReact(
    <Link href={`/slug?slug=${productSearchHit.product_path}`} as={productSearchHit.product_path}>
      <Image
        className='c-product-listing-card__image u-image-shadow'
        src={productSearchHit.product_assets[0].url}
        alt={productSearchHit.product_title}
        aria-label={productSearchHit.product_title}
      />
    </Link>
  )
  expect(wrapper).toIncludeText(productSearchHit.product_title)
  expect(wrapper.find('.c-product-price')).toIncludeText(`Â£${productSearchHit.variant_meta_data.eu.price}`)
  expect(wrapper.find('.c-product-listing-card__rating')).toHaveClassName('c-product-listing-card__rating')
})
