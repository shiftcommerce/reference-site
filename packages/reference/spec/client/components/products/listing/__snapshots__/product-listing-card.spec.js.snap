// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductListingCard correctly 1`] = `
<ProductListingCard
  assetFileAltText="SEPIA nonslip tray 420×210mm"
  assetFileUrl="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
  maxPrice={97.68}
  minPrice={97.68}
  productPath="/products/seed_product_51"
  productRating={0}
  title="SEPIA nonslip tray 420×210mm"
>
  <div
    className="c-product-listing-card"
  >
    <div
      className="c-product-listing-card__body"
    >
      <Link
        as="/products/seed_product_51"
        href="/slug?slug=/products/seed_product_51"
      >
        <a
          href="/products/seed_product_51"
          onClick={[Function]}
        >
          <div
            className="c-product-listing-card__gallery c-image"
          >
            <Image
              alt="SEPIA nonslip tray 420×210mm"
              aria-label="SEPIA nonslip tray 420×210mm"
              className="c-product-listing-card__image u-image-shadow"
              src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
            >
              <div
                className="c-product-listing-card__image u-image-shadow"
              >
                <picture>
                  <img
                    alt="SEPIA nonslip tray 420×210mm"
                    aria-label="SEPIA nonslip tray 420×210mm"
                    className="c-image"
                    src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
                    srcSet="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg 769w"
                  />
                </picture>
              </div>
            </Image>
          </div>
          <div
            className="c-product-listing-card__title"
          >
            <p
              aria-level="1"
              role="heading"
            >
              SEPIA nonslip tray 420×210mm
            </p>
          </div>
        </a>
      </Link>
      <div
        className="c-product-listing-card__price"
      >
        <ProductPrice
          maxPrice={97.68}
          minPrice={97.68}
        >
          <div
            className="c-product-price"
          >
            <span>
              £
              97.68
            </span>
          </div>
        </ProductPrice>
      </div>
      <div
        className="c-product-listing-card__rating"
      >
        <span
          className="c-product-listing-card__rating--empty"
          key="1"
        >
          ★
        </span>
        <span
          className="c-product-listing-card__rating--empty"
          key="2"
        >
          ★
        </span>
        <span
          className="c-product-listing-card__rating--empty"
          key="3"
        >
          ★
        </span>
        <span
          className="c-product-listing-card__rating--empty"
          key="4"
        >
          ★
        </span>
        <span
          className="c-product-listing-card__rating--empty"
          key="5"
        >
          ★
        </span>
      </div>
    </div>
  </div>
</ProductListingCard>
`;
